core/DCache.cpp

```diff
@@ -78,17 +81,21 @@ namespace olympia {
             if(!busy_) {
                 busy_ = true;
                 cache_pending_inst_ = memory_access_info_ptr;
-                out_biu_req_.send(cache_pending_inst_->getInstPtr());
+                out_l2cache_req_.send(cache_pending_inst_->getInstPtr());
             }
         }
         out_lsu_lookup_ack_.send(memory_access_info_ptr);
     }
 
-    void DCache::getAckFromBIU_(const InstPtr &inst_ptr) {
+    void DCache::getRespFromL2Cache_(const InstPtr &inst_ptr) {
         out_lsu_lookup_req_.send(cache_pending_inst_);
         reloadCache_(inst_ptr->getRAdr());
         cache_pending_inst_.reset();
         busy_ = false;
     }
 
+    void DCache::getAckFromL2Cache_(const bool &ack) {
+        // Process ACK
+    }
```

Comment:
What do expect the DCache to do with the acknodelgement?  (in the meantime consider adding a `(void) ack;` to prevent some compilers from complaining about unused parameter).

Reply:
Ack Implementation: When DCache sends the request to L2Cache for a miss, this boolean will be set to false, and Dcache should wait for ack from L2Cache, notifying DCache, that there is "**_space in its dcache request buffer_**".
 
Set `dcache_l2cache_credit_available_` to true so that the following misses from DCache can be sent out to L2Cache.

Reply:
L2Cache _now_ returns 'n' credits based on the number of empty slots available in the requestor port input buffers. DCache should use this number to forward the requests to L2Cache as L2Cache will be able to hold that many number of pending requests from DCache.

*ack_ is now of type uint32_t.